all: require go1.10 to build CockroachDB

nvb opened the pull request:
We just had our first code that requires go1.10 land on master (#23425),
so CockroachDB can no longer be built using go1.9. This change updates
the `go-version-check` script to enforce this.

Release note (build change): Go 1.10 is now the minimum required version
necessary to build CockroachDB.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23441)
<!-- Reviewable:end -->

nvanbenschoten commented:
If we don't want to merge this now then we'll need to revert #23441.

benesch reviewed (commented):
PR (obviously) looks good, assuming Ben is on board with the bump!

mjibson commented:
Also sorry about that PR before this. I honestly didn't know that was a Go 1.10 feature. Definitely agree we should bump this along with the builder version at the same time in the future.

benesch commented:
> Definitely agree we should bump this along with the builder version at the same time in the future.

Yep, that or a separate CI build that verifies compilation still works with the old version of Go.

petermattis commented:
Can we add a comment to `build/builder/Dockerfile` indicating the care that needs to be taken when bumping the Go major version?

benesch commented:
Yep. I think we should add a checklist in docs/, to be copy/pasted into the
PR description, and reference it from build/go-version-check and
build/builder/Dockerfile.

On Tue, Mar 6, 2018 at 9:48 AM Peter Mattis <notifications@github.com>
wrote:

> Can we add a comment to build/builder/Dockerfile indicating the care that
> needs to be taken when bumping the Go major version?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/23441#issuecomment-370805132>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA15IG9hTwkDCvphWhXhRzW9fyoLcyLZks5tbqHOgaJpZM4Sd59T>
> .
>

RaduBerinde commented:
LGTM

Regarding the question of whether we want to support 1.9 or not, note that there was a (minor) gofmt change in 1.10; this might add to the headaches of supporting both versions.

benesch commented:
@nvanbenschoten I think we should merge this as-is. I filed https://github.com/cockroachdb/cockroach/issues/23481 about adding a checklist.

nvb committed:
all: require go1.10 to build CockroachDB

We just had our first code that requires go1.10 land on master (#23425),
so CockroachDB can no longer be built using go1.9. This change updates
the `go-version-check` script to enforce this.

Release note (build change): Go 1.10 is now the minimum required version
necessary to build CockroachDB.

nvb committed:
build: add warning to builder/Dockerfile about bumping Go version

Release note: None

nvb committed:
build: bump Go version in gceworker to Go 1.10

Release note: None

nvanbenschoten commented:
@pmamatsis @benesch I added the warning to `builder/Dockerfile` and also bumped the version of Go installed in new gceworkers. I'll merge on green.

benesch commented:
Still LGTM, though looks like you'll need a rebase.

nvanbenschoten commented:
Where did we leave off with Go 1.10 and the release 2.0 branch? If we're planning to build the release with 1.10 then this should all be cherry-picked over to that branch.

petermattis commented:
The 2.0 branch is building with go1.10. We should cherry-pick this PR.

benesch commented:
For posterity: this was cherrypicked as https://github.com/cockroachdb/cockroach/pull/23494.
